dotnet/src/SemanticKernel.IntegrationTests/Planning/FunctionFlowPlannerTests.cs

```diff
@@ -4,121 +4,68 @@
+    public FunctionFlowPlannerTests(ITestOutputHelper output)
     {
-        this._logger = NullLogger.Instance; //new XunitLogger<object>(output);
+        this._logger = new XunitLogger<object>(output);
         this._testOutputHelper = new RedirectOutput(output);
 
         // Load configuration
         this._configuration = new ConfigurationBuilder()
             .AddJsonFile(path: "testsettings.json", optional: false, reloadOnChange: true)
             .AddJsonFile(path: "testsettings.development.json", optional: true, reloadOnChange: true)
             .AddEnvironmentVariables()
-            .AddUserSecrets<PlanTests>()
+            .AddUserSecrets<FunctionFlowPlannerTests>()
             .Build();
     }
 
     [Theory]
-    [InlineData("Write a poem or joke and send it in an e-mail to Kai.")]
-    public void CreatePlan(string prompt)
-    {
-        // Arrange
-
-        // Act
-        var plan = new Plan(prompt);
-
-        // Assert
-        Assert.Equal(prompt, plan.Description);
-        Assert.Equal(prompt, plan.Name);
-        Assert.Equal("Microsoft.SemanticKernel.Orchestration.Plan", plan.SkillName);
-        Assert.Empty(plan.Steps);
-    }
-
-    [Theory]
-    [InlineData("This is a story about a dog.", "kai@email.com")]
-    public async Task CanExecuteRunSimpleAsync(string inputToEmail, string expectedEmail)
+    [InlineData("Write a poem and send it in an e-mail to Kai.", "SendEmailAsync", "_GLOBAL_FUNCTIONS_")]
+    public async Task CreatePlanFunctionFlowAsync(string prompt, string expectedFunction, string expectedSkill)
     {
         // Arrange
         IKernel target = this.InitializeKernel();
-        var emailSkill = target.ImportSkill(new EmailSkillFake());
-        var expectedBody = $"Sent email to: {expectedEmail}. Body: {inputToEmail}".Trim();
 
-        var plan = new Plan(emailSkill["SendEmailAsync"]);
+        var planner = new FunctionFlowPlanner(target);
 
         // Act
-        var cv = new ContextVariables();
-        cv.Update(inputToEmail);
-        cv.Set("email_address", expectedEmail);
-        var result = await target.RunAsync(cv, plan);
-
+        var plan = await planner.CreatePlanAsync(prompt);
         // Assert
-        Assert.Equal(expectedBody, result.Result);
+        Assert.Contains(
+            plan.Steps,
+            step =>
+                step.Name.Equals(expectedFunction, StringComparison.OrdinalIgnoreCase) &&
+                step.SkillName.Equals(expectedSkill, StringComparison.OrdinalIgnoreCase));
     }
 
     [Theory]
-    [InlineData("Send a story to kai.", "This is a story about a dog.", "French", "kai@email.com")]
-    public async Task CanExecuteRunSimpleStepsAsync(string goal, string inputToTranslate, string language, string expectedEmail)
+    [InlineData("Write a poem or joke and send it in an e-mail to Kai.", "SendEmailAsync", "_GLOBAL_FUNCTIONS_")]
+    public async Task CreatePlanGoalRelevantAsync(string prompt, string expectedFunction, string expectedSkill)
```

Comment:
Ensure this passes with GPT3.

Reply:
Green: https://github.com/lemillermicrosoft/semantic-kernel/actions/runs/4748412995